src/marshaller/HipieDDL.js

```diff
@@ -1083,26 +1086,20 @@
         };
 
         var context = this;
-        transport
-            .call(request, function (response) {
-                if (exists(hipieResultName, response)) {
-                    transport.fetchResult(hipieResultName, function (ddlResponse) {
-                        var json = ddlResponse[0][hipieResultName];
-                        //  Temp Hack  ---
-                        var ddlParts = json.split("<RoxieBase>\\");
-                        if (ddlParts.length > 1) {
-                            var urlEndQuote = ddlParts[1].indexOf("\"");
-                            ddlParts[1] = ddlParts[1].substring(urlEndQuote);
-                            json = ddlParts.join(url);
-                        }
-                        //  ---  ---
-                        context.parse(json, function () {
-                            callback(response);
-                        });
+        transport.call(request).then(function (response) {
+            if (exists(hipieResultName, response)) {
+                return transport.fetchResult(hipieResultName).then(function (ddlResponse) {
+                    var json = ddlResponse[0][hipieResultName];
+                    context.parse(json, function () {
+                        callback(response);
                     });
-                }
-            })
-        ;
+                }).catch(function (e) {
+                    alert(e);
+                });
+            }
+        }).catch(function (e) {
+            alert(e);
```

Comment:
Adding the callback is a good idea.  I am not sure if alert is the best default behavior even, but is probably better than a console log at this point in time...